### PR TITLE
fix return check of some functions

### PR DESCRIPTION
**Pull request by PeiweiHu**
Some fragmentary api return check changes are gathered in this pr. 

It may take effort to review. Thank you for your time, sincerely. 

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by mattcaswell** [comment]
CI failures look relevant

File: test/threadstest.c [hunk 1]
```diff
@@ -437,7 +437,7 @@ static void thread_shared_evp_pkey(void)
             goto err;
 
         if (!TEST_int_ge(EVP_PKEY_decrypt_init(ctx), 0)
-                || !TEST_int_ge(EVP_PKEY_decrypt(ctx, ptbuf, &ptlen, ctbuf, ctlen),
+                || !TEST_int_gt(EVP_PKEY_decrypt(ctx, ptbuf, &ptlen, ctbuf, ctlen),
```
Comment (t8m) [inline test/threadstest.c:440]: ptlen needs to be reinitialized to sizeof(ptbuf) every time.

Comment (t8m) [reply test/threadstest.c:440]: @PeiweiHu can you please fix that?

Comment (PeiweiHu) [reply test/threadstest.c:440]: ok, sorry for being late. I will do it now.

File: test/threadstest.c [hunk 2]
```diff
@@ -436,6 +436,7 @@ static void thread_shared_evp_pkey(void)
         if (!TEST_ptr(ctx))
             goto err;
 
+        ptlen = sizeof(ptbuf);
```
Comment (t8m) [inline test/threadstest.c:439]: Perhaps also move the declaration inside the for loop? Or drop the initial assignment at the declaration as it is useless.

Comment (PeiweiHu) [reply test/threadstest.c:439]: done, thank u for ur reminder.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
I recommend using `git commit --fixup <commit-id>` to commit the fixup commits next time. It makes it easier to do the merge later.

**Comment by PeiweiHu** [comment]
> I recommend using `git commit --fixup <commit-id>` to commit the fixup commits next time. It makes it easier to do the merge later.

Ok, I didn't know this before. I will try next time.   :-)

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master and 3.0 branches. Thank you for your contribution.

**Pull request closed without merging**